### PR TITLE
fix: multiple rds create tests - sc-7578

### PR DESCRIPTION
**Pull request by madhuravius**
Some of these images are dated but this still works with the changes.  I will dupe the tests and adjust so that it will easily run multiple versions in an immediate copy of this PR, i'll probably also remove some repetition then as a result

---

Does create multiple with the same vpc source:

image

image

image

image

image

---

image

---

This is for the approach given the intensity of options, may want to increase worker parallelization

### CODE REVIEW

**Comment by madhuravius** [comment]
drop all the temporary lock files

File: test/basic/rds_create/vpc/create_vpc_for_tests.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	Vars: map[string]interface{}{
+		"organization_id": os.Getenv("ORGANIZATION_ID"),
+		"environment_id":  os.Getenv("ENVIRONMENT_ID"),
+		"aptible_host":    os.Getenv("APTIBLE_HOST"),
+		"vpc_name":        "rds-create-vpc",
+	},
+}
+
+// AcquireVPCOrCreate - will create a VPC with a lock of its existence
+func AcquireVPCOrCreate(t *testing.T, lockName string) error {
```
Comment (madhuravius) [inline test/basic/rds_create/vpc/create_vpc_for_tests.go:23]: turn this into an sdk call and do a get or create

File: test/basic/rds_create/mariadb/rds_mariadb_create_test.go [hunk 2]
```diff
@@ -21,19 +20,26 @@ func cleanupAndAssert(t *testing.T, terraformOptions *terraform.Options) {
...
+			"environment_id":          os.Getenv("ENVIRONMENT_ID"),
+			"aptible_host":            os.Getenv("APTIBLE_HOST"),
+			"database_name":           "test-create-mariadb-57",
+			"database_engine_version": "5.7",
+			"vpc_name":                "rds-create-vpc-mariadb-57",
 		},
 	})
 	defer cleanupAndAssert(t, terraformOptions)
+
+	err := rds_create.CheckOrRequestVPCLimit()
```
Comment (aguilinger) [inline test/basic/rds_create/mariadb/rds_mariadb_create_test.go:38]: do we not want this in an init function thing?

**Comment by aguilinger** [comment]
do we not want to check the secret that gets created alongside these dbs to ensure they (a) exist and (b) are of a form we expect (just regex - making sure they function can be out of scope)

File: test/basic/rds_create/mariadb/rds_mariadb_create_test.go (hunk 2 above)
Comment (madhuravius) [reply test/basic/rds_create/mariadb/rds_mariadb_create_test.go:38]: yeah probably better to do, good call. i can do that inside of that other package i think

File: test/basic/rds_create/mysql/rds_mysql_create_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	assert.Nil(t, rdsErr)
+	assert.Equal(t, rdsAsset.Id, rdsId)
+	assert.Equal(t, rdsAsset.Status, cac.ASSETSTATUS_DEPLOYED)
+	assert.NotNil(t, rdsAsset.Outputs)
+	assert.Equal(t, rdsAsset.GetOutputs()["db_identifier"].Data.(string), rdsInstanceId)
+
+	// check aws asset state
+	rdsAws, rdsAwsErr := terratest_aws.GetRdsInstanceDetailsE(t, rdsInstanceId, "us-east-1")
+	assert.Nil(t, rdsAwsErr)
+	assert.Equal(t, *rdsAws.DBInstanceStatus, "available")
```
[the remark this answers is not in the document]
Comment (madhuravius) [reply test/basic/rds_create/mysql/rds_mysql_create_test.go:70]: sure, i can def do that, wasn't really sure as i mostly pattern matched off what we worked on as a group and we stopped here, but i think that's a good point. i might poke the group as to what we want to validate here

**Comment by madhuravius** [comment]
mariadb is now unblocked on backend with tests passing

image

**Review by aguilinger — DISMISSED** [review]
🎉

**Comment by madhuravius** [comment]
I have to edit the existing test suite to support these 3 sets. For now it's only 3 top level tiers of postgres, but I planned on adding a few versions of each in a follow up PR, so this may balloon a fair amount

**Review by aguilinger — APPROVED** [review]

**Pull request merged**